### PR TITLE
Expose ILogger creation on TaskOrchestrationContext

### LINKED ISSUE #46: Provide replay-safe ILogger in orchestrations
At the moment, the orchestrator is created as a singleton using parameterless constructor. 

```cs
static readonly ITaskOrchestrator singletonOrchestrator = new DurablePocoPayloadSample.Orchestrator();
```

Would it be possible to create it using ActivatorUtilities.CreateInstance instead so we could inject ILogger, IOptions and the like?

### PR DESCRIPTION
**Pull request by jviau**
- Allows for `ILogger` creation from `TaskOrchestrationContext`
- Removes `CreateReplaySafeLogger(ILogger logger)`, as it can now be created directly.
    - In the Functions world, the `FunctionsContext.CreateLogger` methods are basically identical to the `CreateReplaySafeLogger` introduced here, so this method was now redundant.

resolves #46

### CODE REVIEW

**Review by cgillum — COMMENTED** [review]
Looks like there are some build warnings about XML doc comments [here](https://github.com/microsoft/durabletask-dotnet/pull/106/files/0dd1d08a41bf67bbd1c58c107380680c04471c39..f3c3bdd976e40adce9fc9d2a52d341e40f3493f7#file-src-abstractions-taskorchestrator-cs-L85).

File: src/Abstractions/TaskOrchestrationContext.cs [hunk 1]
```diff
@@ -364,18 +369,28 @@ public async Task<T> WaitForExternalEvent<T>(string eventName, TimeSpan timeout)
...
+    public virtual ILogger CreateReplaySafeLogger(Type type)
+        => new ReplaySafeLogger(this, this.LoggerFactory.CreateLogger(type));
+
+    /// <summary>
+    /// Returns an instance of <see cref="ILogger"/> that is replay-safe, meaning that the logger only
+    /// writes logs when the orchestrator is not replaying previous history.
+    /// </summary>
+    /// <typeparam name="T">The type to derive category name from.</typeparam>
+    /// <returns>An instance of <see cref="ILogger"/> that is replay-safe.</returns>
+    public virtual ILogger CreateReplaySafeLogger<T>()
```
Comment (cgillum) [inline src/Abstractions/TaskOrchestrationContext.cs:392]: Consider using `inheritdoc` to remove the redundant `<summary>` and `<returns>` tags.

**Review by cgillum — APPROVED** [review]
This is a nice simplification!

**Comment by elias-rod** [comment]
Hello, when is this going to be available?

**Pull request merged**